### ISSUE TITLE
Poor error message for ambiguous_extension_member_access when two extensions exist with the same name.

### ISSUE THREAD

**Issue by bsutton** [dart-lang/linter#5022]
**Describe the issue**
In my local project I have an extension:

```dart
extension StringAsProcess on String {
  /// run
  Future<void> get run async {
    await HalfPipe().commandAndArgs(this).print();
  }
```
My project also imports the dcli project which has a similar extension
```dart
extension StringAsProcess on String {
  void get run {
    cmd.start(
      this,
      terminal: true,
      progress: Progress(print, stderr: printerr),
    );
  }
```

When I tried to use the 'run' method in a unit test:
```dart
import 'package:dcli/dcli.dart' as dcli;
import '../../../lib/src/shorthand.dart';
import 'package:test/test.dart';

void main() {
  test('stderr stream ...', () async {
    'dart /path/to/exc -e 10'.run;
  });
}

```

The linter reported the error:
```
ambiguous_extension_member_access
 "A member named 'run' is defined in extension 'StringAsProcess' and extension 'StringAsProcess', and none are more specific.\nTry using an extension override to specify the extension you want to be chosen.",
```

**Expected behavior**
The linter should provide details on what library each of the extensions was included from.

```
 "A member named 'run' is defined in extension 'StringAsProcess' from package dcli/src/string_as_process.dart and local extension 'StringAsProcess'  from lib/src/shorthand.dart, and none are more specific.\nTry using an extension override to specify the extension you want to be chosen.",
```

**additional context**
This error was particularly confusing as I had forgotten about the packages local extension and when I aliased the dcli import (as dcli) I expect that to remove the error but I guess extensions bypass the alias?

**Comment by bwilkerson** [comment]
For anyone looking to fix this bug, there's already support for this kind of disambiguation when passing types in as the arguments to the `reportError` methods, but we'll likely need something similar for passing in elements (because extensions don't define types).
